src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -58,76 +54,59 @@ public ActionResult execute() throws EntityDoesNotExistException {
             statusToUser.add(Const.StatusMessages.STUDENT_PROFILE_UNACCESSIBLE_TO_INSTRUCTOR);
         } else {
-            data.studentProfile = logic.getStudentProfile(data.student.googleId);
-            Assumption.assertNotNull(data.studentProfile);
+            studentProfile = logic.getStudentProfile(student.googleId);
+            Assumption.assertNotNull(studentProfile);
         }
 
-        // call this function right before the 'no-records' check
-        loadStudentProfile();
-
-        if (data.sessions.size() == 0 && data.comments.size() == 0) {
+        if (sessions.size() == 0 && comments.size() == 0) {
             statusToUser.add(Const.StatusMessages.INSTRUCTOR_NO_STUDENT_RECORDS);
         }
 
+        List<String> sessionNames = new ArrayList<String>();
+        for (FeedbackSessionAttributes fsa : sessions) {
+            sessionNames.add(fsa.feedbackSessionName);
+        }
+        
+        data = new InstructorStudentRecordsPageData(account, student, courseId, showCommentBox, studentProfile,
+                                                    comments, sessionNames, instructor);
+
         statusToAdmin = "instructorStudentRecords Page Load<br>"
                       + "Viewing <span class=\"bold\">" + studentEmail + "'s</span> records "
                       + "for Course <span class=\"bold\">[" + courseId + "]</span><br>"
-                      + "Number of sessions: " + data.sessions.size() + "<br>"
-                      + "Student Profile: " + (data.studentProfile == null ? "No Profile"
-                                                                           : data.studentProfile.toString());
+                      + "Number of sessions: " + sessions.size() + "<br>"
+                      + "Student Profile: " + (studentProfile == null ? "No Profile"
+                                                                      : studentProfile.toString());
 
         return createShowPageResult(Const.ViewURIs.INSTRUCTOR_STUDENT_RECORDS, data);
     }
 
-    private void filterFeedbackSessions(String courseId, List<FeedbackSessionAttributes> feedbacks) {
+    private void filterFeedbackSessions(String courseId, List<FeedbackSessionAttributes> feedbacks,
+                                        InstructorAttributes instructor, StudentAttributes student) {
         Iterator<FeedbackSessionAttributes> iterFs = feedbacks.iterator();
         while (iterFs.hasNext()) {
             FeedbackSessionAttributes tempFs = iterFs.next();
             if (!tempFs.courseId.equals(courseId)) {
                 iterFs.remove();
-            } else if (!data.currentInstructor.isAllowedForPrivilege(data.student.section, 
-                    tempFs.getSessionName(), Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS)) {
+            } else if (!instructor.isAllowedForPrivilege(student.section, tempFs.getSessionName(),
```

Comment:
Since they're both doing an `iterFs.remove()`, should this be an `||` rather than an `if-else-if`?